src/shogun/lib/any.h

```diff
@@ -273,6 +270,14 @@ namespace shogun {
 			return compare(lhs(), rhs());
 		}
 
+		template <>
+		bool compare_impl<void>(
+				maybe_most_important, const std::function<void()>& lhs,
+				const std::function<void()>& rhs)
+		{
+			return true;
```

Comment:
@karlnapf for now I put `std::function<void()> == std::function<void()>` is true, but can argue either way tbh..